sandboxify.c

```diff
@@ -25,89 +27,144 @@ static void usage(const char *prog)
 
 int main(int argc, char **argv)
 {
+	int pipefds[2];
+	char data = '\0';
 	pid_t child = 0;
+	pid_t parent = getpid();
 
 	if (argc < 2) {
 		usage(basename(argv[0]));
 		return EXIT_FAILURE;
 	}
 
+	/* Could use pipe2 */
+	if (pipe(pipefds)) {
+		perror("Failed to setup pipes");
+		return EXIT_FAILURE;
+	}
+	fcntl(pipefds[0], F_SETFD, FD_CLOEXEC);
```

Comment:
According to https://man7.org/linux/man-pages/man2/pipe.2.html we could replace these with a flag `O_CLOEXEC` to `pipe2` call above (and use `pipe2` above as well).

The code will be a bit smaller.

Reply:
That was what I was thinking. I had it as a comment but I'll do it now

Reply:
Added the `_GNU_SOURCES` define and switched over to `pipe2()`